props/saber_fett263_buttons.h

```diff
@@ -285,47 +440,1090 @@ EFFECT(faston); // for EFFECT_FAST_ON
+              SFX_mnum.Select(say2_ - 1);
+              wav_player->PlayOnce(&SFX_mnum, 0.0);
+              play_number_ = PLAY_HUNDRED;
+              break;            
+            case PLAY_HUNDRED:
+              PlayMenuSound("hundred.wav");
+              play_number_ = PLAY_NONE;
+              say_type_ = SAY_NONE;
+              wav_player.Free();              
+            default:  
+              play_number_ = PLAY_NONE;
+              say_type_ = SAY_NONE;
+              break;
+          } 
+          break;
+        default:
+          break;
+      }
+    }
+#ifdef FETT263_EDIT_MODE_MENU      
+    if (next_event_ && !wav_player->isPlaying()) {
+      switch (menu_type_) {
+        case MENU_IGNITION_TIME:
+        case MENU_RETRACTION_TIME:
+          next_event_ = false;
+          SetInOut();
+          break;
+        default:
+          next_event_ = false;
+          break;
+      }
+    }
+    if (off_event_ && millis() - last_rotate_ > 200) {
+      Off();
+      off_event_ = false;
+      restart_ = true;
+      last_rotate_ = millis();
+    }
+    if (restart_ && millis() - last_rotate_ > calc_ + 1000) {
+      restart_ = false;
+      FastOn();
+    }
+#endif
+    if (announce_menu_ && !wav_player->isPlaying()) {
+      if (rehearse_) {
+        if (SaberBase::IsOn()) {
+          rehearse_ = false;
+          Off();
+#ifdef FETT263_SAVE_CHOREOGRAPHY
+          SaveChoreo();
+#endif
+        } else {
+          PlayMenuSound("rehrsbgn.wav");
+          FastOn();
+        }
+      } else {
+#ifdef FETT263_EDIT_MODE_MENU        
+        switch (menu_type_) {
+          case MENU_TOP:
+            PlayMenuSound("mmain.wav");
+            break;
+          case MENU_COLOR_SUB:
+            PlayMenuSound("mcolorsb.wav");
+            break;
+          case MENU_COLOR_MODE:
+            PlayMenuSound("mcolorop.wav");
+            break;
+#if (NUM_BLADES > 1)
+          case MENU_BLADE_COLOR:
+            PlayMenuSound("mblade.wav");
+            break;
+          case MENU_BLADE_STYLE:
+            PlayMenuSound("mblade.wav");
+            break;
+          case MENU_BLADE_LENGTH:
+            PlayMenuSound("mblade.wav");
+            break;
+#endif
+          case MENU_GESTURE_SUB:
+            PlayMenuSound("mgestsub.wav");
+            break;
+          case MENU_SETTING_SUB:
+            PlayMenuSound("msetsub.wav");
+            break;
+          case MENU_EFFECT:
+            PlayMenuSound("meffect.wav");
+            break;
+          case MENU_STYLE_SETTING_SUB:
+            PlayMenuSound("mstylstm.wav");
+            break;
+        }
+#endif
+      }
+        announce_menu_ = false;  
+    } 
+
+#ifdef FETT263_EDIT_MODE_MENU
+      if (edit_color_ && color_mode_ != COLOR_LIST) {
+#define H_CHANGE (M_PI / 98304)
+#define L_ANGLE (M_PI / 30)
+#define H_ANGLE (M_PI / 16384)
+#define EDIT_MODE_ZOOM (M_PI * 2 / 2000)
+      float a = fusor.angle2() - hsl_angle_;
+      switch (color_mode_) {
```

Comment:
All of these switch statements look fairly similar. Is there a way we can deduplicate some of the code?
In particular, the first two if statements looks the same in each case, so we could just run that before the switch?

Reply:
I was trying, there are small but significant differences in each case but trying to dissect was getting messier for me.  The cases have different angles for detection and/or are changing different values within different ranges; L_ANGLE, H_ANGLE, EDIT_MODE_ZOOM as well as hsl.H or hsl.L (split 0.0 ~ 0.5 / 0.5 ~ 1.0).  I'm open to suggestions but had been trying and it either stopped working, created new bugs or returned compile errors in my attempts thus far.

Reply:
Well all seem to start with:
       if (a > M_PI) a-=M_PI*2;      
       if (a < -M_PI) a+=M_PI*2;
so maybe we can do that before the switch?

Reply:
Also, they all seem to end with ShowCOlorStyle::SetColor(Color16(hsl)); so maybe we can do that after the switch?

Reply:
ah, makes sense, updated